crates/net/network/src/manager.rs

```diff
@@ -343,6 +345,19 @@ where
         self.swarm.state().peers().handle()
     }
 
+    /// Collect the peers from the [`NetworkManager`] and write them to the given
+    /// `persistent_peers_file`, if configured.
+    pub fn write_peers_to_file(&self, persistent_peers_file: Option<PathBuf>) -> eyre::Result<()> {
+        if let Some(file_path) = persistent_peers_file {
```

Comment:
okay, looking at this function, we can improve a few things here:

1. make the argument not optional, this should be checked by the caller because if none this is a noop
2. then we can also log the the trace! for reth:: cli on the callside
3. this should not return eyre:: and instead the fspath error

Reply:
agree with all above three pointers, very good best practices to stick to in general